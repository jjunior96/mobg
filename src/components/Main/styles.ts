import styled, { css } from 'styled-components';

export const Container = styled.div`
  display: flex;

  width: 100%;
  height: 100vh;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.grid.container};

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  `}
`;
