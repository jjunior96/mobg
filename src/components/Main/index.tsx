import * as S from './styles';

interface MainProps {
  children: React.ReactNode;
}

const Main = ({ children }: MainProps) => (
  <S.Container>
    <S.Content>{children}</S.Content>
  </S.Container>
);

export default Main;
