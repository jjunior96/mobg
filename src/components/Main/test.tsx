import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import Main from '.';

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = renderWithTheme(
      <Main>
        <h1>Next Example</h1>
      </Main>
    );

    expect(
      screen.getByRole('heading', { name: /Next Example/i })
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
