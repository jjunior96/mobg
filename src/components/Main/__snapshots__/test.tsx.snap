// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100vh;
}

.c1 {
  width: 100%;
  max-width: 130rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Next Example
      </h1>
    </div>
  </div>
</div>
`;
